src/backends/simple.ts

```diff
@@ -0,0 +1,442 @@
+import type {
+    DecodedFrames,
+    Packet,
+    Demuxer,
+    Decoder,
+    Filterer
+} from '@antoinemopa/beamcoder';
+import beamcoder from '@antoinemopa/beamcoder';
+import type { ImageData } from 'canvas';
+import { createImageData } from 'canvas';
+import { BaseExtractor } from '../BaseExtractor';
+import type { Extractor, ExtractorArgs, InterpolateMode } from '../../framefusion';
+import { DownloadVideoURL } from '../DownloadVideoURL';
+
+const LOG = false;
+
+const createDecoder = ({
+    demuxer,
+    streamIndex,
+    threadCount,
+}: {
+    demuxer: Demuxer;
+    streamIndex: number;
+    threadCount: number;
+}): Decoder => {
+    return beamcoder.decoder({
+        demuxer: demuxer,
+        width: demuxer.streams[streamIndex].codecpar.width,
+        height: demuxer.streams[streamIndex].codecpar.height,
+        stream_index: streamIndex,
+        pix_fmt: demuxer.streams[streamIndex].codecpar.format,
+        thread_count: threadCount,
+    });
+};
+
+const createFilter = async({
+    stream,
+    outputPixelFormat,
+    interpolateFps,
+    interpolateMode = 'fast',
+}: {
+    stream: beamcoder.Stream;
+    outputPixelFormat: string;
+    interpolateFps?: number;
+    interpolateMode?: InterpolateMode;
+}): Promise<beamcoder.Filterer> => {
+    if (!stream.codecpar.format) {
+        return null;
+    }
+
+    let filterSpec = [`[in0:v]format=${stream.codecpar.format}`];
+
+    if (interpolateFps) {
+        if (interpolateMode === 'high-quality') {
+            filterSpec = [...filterSpec, `minterpolate=fps=${interpolateFps}`];
+        }
+        else if (interpolateMode === 'fast') {
+            filterSpec = [...filterSpec, `fps=${interpolateFps}`];
+        }
+        else {
+            throw new Error(`Unexpected interpolation mode: ${interpolateMode}`);
+        }
+    }
+
+    filterSpec = [...filterSpec];
+
+    const filterSpecStr = filterSpec.join(', ') + '[out0:v]';
+
+    LOG && console.log(`filterSpec: ${filterSpecStr}`);
+
+    return beamcoder.filterer({
+        filterType: 'video',
+        inputParams: [
+            {
+                name: 'in0:v',
+                width: stream.codecpar.width,
+                height: stream.codecpar.height,
+                pixelFormat: stream.codecpar.format,
+                timeBase: stream.time_base,
+                pixelAspect: stream.sample_aspect_ratio,
+            },
+        ],
+        outputParams: [
+            {
+                name: 'out0:v',
+                pixelFormat: outputPixelFormat,
+            },
+        ],
+        filterSpec: filterSpecStr,
+    });
+};
+
+const MAX_PACKET_READS = 1000;
```

Comment:
Fair point. I don't know if we want to keep this. I added this to ensure our while look resolves. Since vitest only shows the result after it completes a test I was unable to debug this otherwise.

Reply:
Ahh I see, it's like a safety measure so we don't loop indefinitely

Reply:
I'm fine with removing or keeping, as long as we add a comment!

Reply:
I removed this